### PR TITLE
Fix: allow referencing dynamic and static variables in one field

### PR DESCRIPTION
**Pull request by michaelhtm**
**Issue #139**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/runtime/runtime.go [hunk 1]
```diff
@@ -346,6 +346,7 @@ func (rt *ResourceGroupRuntime) evaluateDynamicVariables() error {
 	// and are resolved after all the dependencies are resolved.
 
 	resolvedResources := maps.Keys(rt.resolvedResources)
+	resolvedResources = append(resolvedResources, "schema")
```
Comment (a-hilaly) [inline internal/runtime/runtime.go:349]: looks we allowed this during the RG parsing layer, but we didn't during the runtime execution layer. Great catch!

**Comment by a-hilaly** [comment]
@michaelhtm looks like we need to inject real unstrcutured instance objects for the unit tests - i'm also wondering if we can have integration tests for this already, if it's not already the case.

File: internal/runtime/runtime_test.go [hunk 2]
```diff
@@ -1690,6 +1690,9 @@ func Test_evaluateDynamicVariables(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			rt := &ResourceGroupRuntime{
+				instance: newTestResource(
+					withObject(map[string]interface{}{}),
+				),
```
Comment (a-hilaly) [inline internal/runtime/runtime_test.go:1695]: I wonder how is this working when object is empty?

Comment (michaelhtm) [reply internal/runtime/runtime_test.go:1695]: It just looks for object to not be nil..it seems like this test does not need object

**Pull request merged**